JSONObject.java

```diff
@@ -578,10 +581,10 @@ public double getDouble(String key) throws JSONException {
         Object object = this.get(key);
         try {
             return object instanceof Number ? ((Number) object).doubleValue()
-                    : Double.parseDouble((String) object);
+                    : new BigDecimal((String) object).doubleValue();
```

Comment:
but then `int` typecast would be wrong for my proposed shortcut but it can be easily mitigated by checking if the double value is between `Integer.MIN_VALUE` and `Integer.MAX_VALUE` and if it isn't throw an exception.